lib/Raven/Context/RuntimeContext.php

```diff
@@ -0,0 +1,123 @@
+
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * This class is a specialized implementation of the {@see `Context`} class
+ * that stores information about the current runtime.
+ *
+ * @author Stefano Arlandini <sarlandini@alice.it>
+ */
+class RuntimeContext extends Context
+{
+    /**
+     * @var OptionsResolver The options resolver
+     */
+    private $resolver;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws UndefinedOptionsException If any of the options are not supported
+     *                                   by the context
+     * @throws InvalidOptionsException   If any of the options don't fulfill the
+     *                                   specified validation rules
+     */
+    public function __construct(array $data = [])
+    {
+        $this->resolver = new OptionsResolver();
+
+        $this->configureOptions($this->resolver);
+
+        parent::__construct($this->resolver->resolve($data));
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws UndefinedOptionsException If any of the options are not supported
+     *                                   by the context
+     * @throws InvalidOptionsException   If any of the options don't fulfill the
+     *                                   specified validation rules
+     */
+    public function merge(array $data, $recursive = false)
+    {
+        $data = $this->resolver->resolve($data);
+
+        parent::merge($data, $recursive);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws UndefinedOptionsException If any of the options are not supported
+     *                                   by the context
+     * @throws InvalidOptionsException   If any of the options don't fulfill the
+     *                                   specified validation rules
+     */
+    public function setData(array $data)
+    {
+        $data = $this->resolver->resolve($data);
+
+        parent::setData($data);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws UndefinedOptionsException If any of the options are not supported
+     *                                   by the context
+     * @throws InvalidOptionsException   If any of the options don't fulfill the
+     *                                   specified validation rules
+     */
+    public function replaceData(array $data)
+    {
+        $data = $this->resolver->resolve($data);
+
+        parent::replaceData($data);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws UndefinedOptionsException If any of the options are not supported
+     *                                   by the context
+     * @throws InvalidOptionsException   If any of the options don't fulfill the
+     *                                   specified validation rules
+     */
+    public function offsetSet($offset, $value)
```

Comment:
Do we need this method? Isn't better to add a simple `setPlatform()` and `setVersion()`? Also getters too would be nice.

Reply:
>Do we need this method?

We need it because it's inherited from the base class, so I have to override it as well and check that what you try to set is allowed by the interface schema definition

>Isn't better to add a simple `setPlatform()` and `setVersion()`? Also getters too would be nice.

Done